x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/esnative/NativeUsersStore.java

```diff
@@ -697,5 +691,13 @@ ReservedUserInfo deepClone() {
         boolean verifyPassword(SecureString data) {
             return hasher.verify(data, this.passwordHash);
         }
+
+        static ReservedUserInfo defaultEnabledUserInfo(){
+            return new ReservedUserInfo(new char[0], true, true);
```

Comment:
With this PR, the empty passwordHash now has a deterministic value, `new char[0]`. It would be better to remove  `hasEmptyPassword` from the constructor parameters of `ReservedUserInfo` because it can be safely derived. It is more consistent to derive it than manually passing it into the constructor. We can also replace the `hasEmptyPassword` field with a method call which internally returns `passwordHash.length == 0`.

Reply:
Makes sense, will do